File: 30_days_of_code/day_10.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    n = int(input().strip())
    rests = []
    count = 0
    max_sequence = 0
    
    if (n == 0 or n == 1):
        print(n)

    while (n / 2 > 0):
        rest = n % 2
        rests.append(rest)
        n = math.floor(n / 2)
        
    for number in rests:
        if (number == 1):
            count +=1
            if (count > max_sequence):
                max_sequence = count
        else:
            count = 0
            
    print(max_sequence)
        

File: hackerrank/30_days_of_code/day10.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}



function main() {
    let n = parseInt(readLine().trim(), 10);
    
    let rests = [];
    let count = 0;
    let maxSequence = 0;
    
    if (n == 0 || n == 1){
        console.log(n);
    }
    
    while (n / 2 > 0){
        let rest = n % 2;
        rests += rest;
        n = Math.floor(n / 2);
    }
    
    for (let number of rests){
        if (number == 1){
            count += 1
            if (count > maxSequence){
                maxSequence = count
            }
        } else {
            count = 0
        }
    }
    
    console.log(maxSequence)
}
